### PR TITLE
WT-6963 Migrate hs verify to use the new history store cursor

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/history/hs_verify.c [hunk 1]
```diff
@@ -59,15 +56,6 @@ __hs_verify_id(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *ds_cbt, uint32_t this_
         if (btree_id != this_btree_id)
```
Comment (bvpvamsikrishna) [inline src/history/hs_verify.c:56]: Do we still need this check. Don't we get the return code as WT_NOTFOUND, if btree_id is not matching?

Comment (quchenhao) [reply src/history/hs_verify.c:56]: Yes, we don't set the btree id in one of the caller.

Comment (hbokhari) [reply src/history/hs_verify.c:56]: I agree - we can actually change this to an assert.

Comment (quchenhao) [reply src/history/hs_verify.c:56]: It just walks the whole hs tree.

Comment (hbokhari) [reply src/history/hs_verify.c:56]: Got it.

**Comment by hbokhari** [comment]
@quchenhao changes look good to me. However, I was wondering if we can test these changes somehow with HS verification enabled and running few test that will not create too much trouble.

**Comment by quchenhao** [comment]
> @quchenhao changes look good to me. However, I was wondering if we can test these changes somehow with HS verification enabled and running few test that will not create too much trouble.

I can create a patch build to try that.

**Comment by quchenhao** [comment]
patch build:
https://spruce.mongodb.com/version/6007e57232f4177371522c48/tasks

**Comment by quchenhao** [comment]
@hbokhari @bvpvamsikrishna  I tried to enable the hs verification, but it is still failing.

**Review by hbokhari — APPROVED** [review]
LGTM!

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**